Related to https://github.com/llvm/llvm-project/issues/109949
Normally run.py does not carry over LD_LIBRARY_PATH, which can cause some tests to fail on system where
libatomic is not present on a standard system path.
--- libcxx/utils/run.py.orig
+++ libcxx/utils/run.py
@@ -55,6 +55,10 @@
             v = v + os.pathsep + os.environ[k]
         env[k] = v
 
+    if any(isTestExe(exe) for exe in commandLine):
+        for var in ['LD_LIBRARY_PATH', ]:
+            env[var] = os.pathsep.join(filter(None, [os.environ.get(var, ''), env.get(var, '')]))
+
     if platform.system() == "Windows":
         # Pass some extra variables through on Windows:
         # COMSPEC is needed for running subprocesses via std::system().
